python/tvm/relay/op/contrib/dnnl.py

```diff
@@ -85,3 +87,76 @@ def pattern_table():
     conv2d_relu_pat = ("dnnl.conv2d_relu", make_pattern(with_bias=False))
     dnnl_patterns = [conv2d_bias_relu_pat, conv2d_relu_pat]
     return dnnl_patterns
+
+
+def AutoQuery(N,IC,KH,KW,OC,SH,SW,PH_L,PH_R,PW_L,PW_R,OH,OW,G):
+    """Get the optimal layout of dnnl, given shape of conv2d.
+
+    Parameters
+    ----------
+    N,IC,KH,KW,OC,SH,SW,PH_L,PH_R,PW_L,PW_R,OH,OW,G : Int
+                                                     Input argument.
+
+    Returns
+    -------
+    layouts : string
+              The result.
+    """
+    return _ffi_api.AutoQuery(N,IC,KH,KW,OC,SH,SW,PH_L,PH_R,PW_L,PW_R,OH,OW,G)
+
+
+def alter_conv2d_for_dnnl(attrs, inputs, tinfos, out_type):
+    """The layout auto-query func for dnnl."""
+    data_dic = {"a":"N",
+                "c":"H",
+                "b":"C",
+                "d":"W",}
+
+    weight_dic = {"a":"O",
+                  "b":"I",
+                  "c":"H",
+                  "d":"W",
+                  "e":"G"}
+
+    data, weight = inputs
+
+    OC, IC, KH, KW = get_const_tuple(weight.type_annotation.shape)
+    N, OC, OH, OW = get_const_tuple(out_type.shape)
+    PH_L, PW_L, PH_R, PW_R = attrs.get_int_tuple("padding")
+    SH, SW = attrs.get_int_tuple("strides")
+    dilation = attrs.get_int_tuple("dilation")
+    G = int(attrs.groups)
+    new_attrs = dict(attrs)
+
+    # To do optimal layout transform for depthwise conv2d.
+    # Set depthwise conv2d as plain format currently.
+    if G>1: 
+        IC = IC * G
```

Comment:
`G>1` should be `group convolution` which is more general than `depthwise conv`. `IC=IC*G` is useless here?
BTW, suppose the layout should be 5D instead of 4D format?

Reply:
IC=IC*G has been removed.